sympy/stats/tests/test_discrete_rv.py

```diff
@@ -261,3 +262,48 @@ def test_product_spaces():
 #    assert str(P(Eq(X1 + X2, 3))) == """Sum(Piecewise((2**(X2 - 2)*(2/3)**(X2 - 1)/6, """ +\
 #        """X2 <= 2), (0, True)), (X2, 1, oo))"""
     assert P(Eq(X1 + X2, 3)) == Rational(1, 12)
+
+
+def test_sampling_methods():
+    distribs_numpy = [
+        Geometric('G', 0.5),
+        Poisson('P', 1),
+        Zeta('Z', 2)
+    ]
+    distribs_scipy = [
+        Geometric('G', 0.5),
+        Logarithmic('L', 0.5),
+        Poisson('P', 1),
+        Skellam('S', 1, 1),
+        YuleSimon('Y', 1),
+        Zeta('Z', 2)
+    ]
+    distribs_pymc3 = [
+        Geometric('G', 0.5),
+        Poisson('P', 1),
+    ]
+    size = 3
+    numpy = import_module('numpy')
+    if not numpy:
+        skip('Numpy is not installed. Abort tests for _sample_numpy.')
+    else:
+        for X in distribs_numpy:
+            sam = X.pspace.distribution._sample_numpy(size)
+            for i in range(size):
+                assert sam[i] in X.pspace.domain.set
+    scipy = import_module('scipy')
+    if not scipy:
+        skip('Scipy is not installed. Abort tests for _sample_scipy.')
+    else:
+        for X in distribs_scipy:
+            sam = X.pspace.distribution._sample_scipy(size)
+            for i in range(size):
+                assert sam[i] in X.pspace.domain.set
+    pymc3 = import_module('pymc3')
+    if not pymc3:
+        skip('PyMC3 is not installed. Abort tests for _sample_pymc3.')
+    else:
+        for X in distribs_pymc3:
+            sam = X.pspace.distribution._sample_pymc3(size)
```

Comment:
Does `sam = sample(X, size)` works?

Reply:
Yes, it works

Reply:
Please update the tests to use something like, `sam = sample(X, size)`.

Reply:
I have used `samps = sample(X, size) `in case of scipy tests and not in others because it covers almost all distribution but on the other hand numpy and pymc3 does not, so for scipy tests I have used `samps = sample(X, size)` and for other two  `samps = X.pspace.distribution._sample_numpy(size)` and `samps = X.pspace.distribution._sample_pymc3(size)`

Reply:
> Please update the tests to use something like, `sam = sample(X, size)`

I also added tests for using size wherever I changed the parameters